src/overview.md

```diff
@@ -240,6 +240,22 @@ More information:
 * [`webdev` tool](/tools/webdev)
 * [Web deployment tips](/web/deployment)
 
+#### The Dart Runtime {#runtime}
+
+Regardless of how you compile your Dart code, a Dart runtime is needed to
+execute the code. This runtime is responsible for a number of critical tasks:
+
+* Managing memory: Dart uses a managed memory model, where unused memory is
+  reclaimed by a garbage collector (GC). 
+
+* Enforcing the Dart type system: While most type checks are static (compile
+  time), some type checks are dynamic (run time), e.g. the Dart
+  [type check and cast operators](/guides/language/language-tour#type-test-operators).
+
```

Comment:
I'd add also a note about isolates which are also part of the runtime system.

Reply:
How about something like this?

```
* Managing concurrency through [isolates](/guides/language/language-tour#isolates),
  the isolation containers that all Dart code runs within.
```

Reply:
Isolates are not units of isolation - that's too strong of a wording. Maybe just "Managing concurrently executing [isolates](...)".